### PR TITLE
fix: fixed draft deletion of filings

### PR DESCRIPTION
**Pull request by JazzarKarim**
ticket #: https://github.com/bcgov/entity/issues/24885

Fixed draft deletion of IA drafts. I added support to the other types as well.

### CODE REVIEW

File: business-registry-dashboard/app/stores/affiliations.ts [hunk 1]
```diff
@@ -42,10 +42,15 @@ export const useAffiliationsStore = defineStore('brd-affiliations-store', () =>
...
-    if (payload.business.corpType.code === CorpTypes.INCORPORATION_APPLICATION) {
+    // If the business is a new IA, amalgamation, continuation in, or registration then remove the business filing from legal-db
+    if ([
+      CorpTypes.INCORPORATION_APPLICATION,
+      CorpTypes.AMALGAMATION_APPLICATION,
+      CorpTypes.CONTINUATION_IN,
+      CorpTypes.REGISTRATION
+    ].includes(payload.business.corpType.code)) {
       const filingResponse = await getFilings(payload.business.businessIdentifier)
-      if (filingResponse && filingResponse.status === 200) { // TODO: fix typing
```
Comment (JazzarKarim) [inline business-registry-dashboard/app/stores/affiliations.ts:48]: This `filingResponse.status` check is not needed as it's not returned. So, the previous code never actually entered this block.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-registry-dev--pr-38-i03yjms3.web.app

File: business-registry-dashboard/app/stores/affiliations.ts [hunk 2]
```diff
@@ -42,10 +42,15 @@ export const useAffiliationsStore = defineStore('brd-affiliations-store', () =>
   }
 
   async function removeBusiness (payload: RemoveBusinessPayload) {
-    // If the business is a new registration then remove the business filing from legal-db
-    if (payload.business.corpType.code === CorpTypes.INCORPORATION_APPLICATION) {
+    // If the business is a new IA, amalgamation, continuation in, or registration then remove the business filing from legal-db
```
Comment (severinbeauvais) [inline business-registry-dashboard/app/stores/affiliations.ts:45]: We want to allow deletion of Continuation In filing only when it's in DRAFT status. After that there's a review record linked to it and we haven't talked about what to do with it, so for now it's not deleteable.

Comment (JazzarKarim) [reply business-registry-dashboard/app/stores/affiliations.ts]: Thanks for catching this. I removed this and added a TODO item to add the continuation in later.

Comment (severinbeauvais) [reply business-registry-dashboard/app/stores/affiliations.ts]: Fine by me. Resolving.

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**